lib/maths/CBoostedTreeLoss.cc

```diff
@@ -144,11 +144,13 @@ CArgMinLogisticImpl::TDoubleVector CArgMinLogisticImpl::value() const {
     // case we only need one pass over the data and can compute the optimal
     // value from the counts of the two categories.
     if (this->bucketWidth() == 0.0) {
-        objective = [this](double weight) {
+        double prediction{(m_PredictionMinMax.min() + m_PredictionMinMax.max()) / 2.0};
```

Comment:
```suggestion
        double predictionSofar{(m_PredictionMinMax.min() + m_PredictionMinMax.max()) / 2.0};
```

Reply:
I would add `Sofar` or something similar to signify that we are talking about the prediction of the forest so far.

Reply:
Because I use `m_PredictionMinMax` I feel like it is more consistent to call this just prediction. I could change the name of this variable, but I think perhaps the best thing to do is just add an explaining comment. See [5a9d840](https://github.com/elastic/ml-cpp/pull/1032/commits/5a9d84009279d5aa635d4f0db21f685219550d0c).